core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -0,0 +1,426 @@
+	maxCallbacks int, // TODO: unused for now
+) (blocks []ocr2vrftypes.Block, callbacks []ocr2vrftypes.AbstractCostedCallbackRequest, err error) {
+	// TODO: use head broadcaster instead?
+	currentHead, err := c.evmClient.HeadByNumber(ctx, nil)
+	if err != nil {
+		err = errors.Wrap(err, "header by number")
+		return
+	}
+	currentHeight := currentHead.Number
+
+	logs, err := c.lp.LogsWithSigs(
+		currentHeight-c.lookbackBlocks,
+		currentHeight-1,
+		[]common.Hash{
+			c.randomnessRequestedTopic,
+			c.randomnessFulfillmentRequestedTopic,
+			c.randomWordsFulfilledTopic,
+			c.newTransmissionTopic,
+		},
+		c.coordinatorAddress,
+		pg.WithParentCtx(ctx))
+	if err != nil {
+		err = errors.Wrap(err, "logs with topics")
+		return
+	}
+
+	var (
+		randomnessRequestedLogs            []vrf_wrapper.VRFBeaconCoordinatorRandomnessRequested
+		randomnessFulfillmentRequestedLogs []vrf_wrapper.VRFBeaconCoordinatorRandomnessFulfillmentRequested
+		randomWordsFulfilledLogs           []vrf_wrapper.VRFBeaconCoordinatorRandomWordsFulfilled
+		newTransmissionLogs                []vrf_wrapper.VRFBeaconCoordinatorNewTransmission
+	)
+	for _, lg := range logs {
+		switch {
+		case bytes.Equal(lg.EventSig, c.randomnessRequestedTopic[:]):
+			unpacked, err2 := unmarshalRandomnessRequested(lg) // FIXME
+			if err2 != nil {
+				// should never happen
+				err = errors.Wrap(err2, "unmarshal RandomnessRequested log")
+			}
+			randomnessRequestedLogs = append(randomnessRequestedLogs, unpacked)
+		case bytes.Equal(lg.EventSig, c.randomnessFulfillmentRequestedTopic[:]):
+			unpacked, err2 := unmarshalRandomnessFulfillmentRequested(lg) // FIXME
+			if err2 != nil {
+				// should never happen
+				err = errors.Wrap(err2, "unmarshal RandomnessFulfillmentRequested log")
+			}
+			randomnessFulfillmentRequestedLogs = append(randomnessFulfillmentRequestedLogs, unpacked)
+		case bytes.Equal(lg.EventSig, c.randomWordsFulfilledTopic[:]):
+			unpacked, err2 := unmarshalRandomWordsFulfilled(lg) // FIXME
+			if err2 != nil {
+				// should never happen
+				err = errors.Wrap(err2, "unmarshal RandomWordsFulfilled log")
+			}
+			randomWordsFulfilledLogs = append(randomWordsFulfilledLogs, unpacked)
+		case bytes.Equal(lg.EventSig, c.newTransmissionTopic[:]):
+			unpacked, err2 := unmarshalNewTransmission(lg) // FIXME
+			if err2 != nil {
+				// should never happen
+				err = errors.Wrap(err2, "unmarshal NewTransmission log")
+			}
+			newTransmissionLogs = append(newTransmissionLogs, unpacked)
+		}
+	}
+
+	// Scan for blocks where an output is required
+	// blocksRequested maps block number to the block object.
+	type key struct {
+		blockNumber uint64
+		confDelay   uint32
+	}
+	blocksRequested := make(map[key]struct{})
+	for _, r := range randomnessRequestedLogs {
+		if _, ok := confirmationDelays[uint32(r.ConfDelay.Uint64())]; !ok {
+			// if we can't find the conf delay in the map then just ignore this request
+			c.lggr.Infow("ignoring bad request, unsupported conf delay",
```

Comment:
Should we change this to Error log instead? Seems like an exceptional scenario since requests with unknown conf delays are [reverted](https://github.com/smartcontractkit/threshold-vrf/blob/master/contracts/VRFBeaconExternalAPI.sol#L317)